packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -300,10 +305,12 @@ class CupertinoTextField extends StatefulWidget {
        assert(readOnly != null),
        assert(autofocus != null),
        assert(obscuringCharacter != null && obscuringCharacter.length == 1),
-       assert(obscureText != null),
+       assert(obscureText != null || obscureTextBehavior != null),
        assert(autocorrect != null),
-       smartDashesType = smartDashesType ?? (obscureText ? SmartDashesType.disabled : SmartDashesType.enabled),
-       smartQuotesType = smartQuotesType ?? (obscureText ? SmartQuotesType.disabled : SmartQuotesType.enabled),
+       smartDashesType = smartDashesType ?? ((obscureText != null && obscureText) ||
+           (obscureTextBehavior != null && obscureTextBehavior != ObscureTextBehavior.none) ? SmartDashesType.disabled : SmartDashesType.enabled),
+       smartQuotesType = smartQuotesType ?? ((obscureText != null && obscureText) ||
```

Comment:
Instead of checking for the existence of obscureText everywhere, what if we convert it to obscureTextBehavior in the initializers and then only use obscureTextBehavior?

```dart
obscureTextBehavior = obscureTextBehavior ?? (obscureText != null && obscureText) ? ObscureTextBehavior.delayed : ObscureTextBehavior.none,
```

Then I think you can pretty much get rid of obscureText everywhere else.